tests/test_utils.py

```diff
@@ -19,14 +19,333 @@
 #     Santiago Dueñas <sduenas@bitergia.com>
 #
 
+import datetime
+
+from dateutil.tz import UTC
+
 from django.test import TestCase
 
-from sortinghat.core.utils import unaccent_string
+from sortinghat.core.utils import merge_datetime_ranges, unaccent_string
 
 
+CANT_COMPARE_DATES_ERROR = "can't compare offset-naive and offset-aware datetimes"
+DATE_OUT_OF_BOUNDS_ERROR = "'{type}' date {date} is out of bounds"
 UNACCENT_TYPE_ERROR = "argument must be a string; int given"
 
 
+class TestMergeDatetimeRanges(TestCase):
+    """Unit tests for merge_datetime_ranges function"""
+
+    def test_merge_datetime_ranges(self):
```

Comment:
The test method `test_merge_datetime_ranges` contains many cases, maybe they could be splitted in different test methods. Another option could be to add some details in the line comments (`# Case 1: ...., `# Case 2: ..., etc.)

Reply:
I mainly followed the approach of commenting each case. It was hard to describe each case. In some cases it was more clear I create new test cases.

Reply:
perfect!